scripts/should_deploy.py

```diff
@@ -0,0 +1,247 @@
+            return c['sha']
+
+
+class Travis:
+    url = "https://api.travis-ci.com"
+
+    def __init__(self, owner, repo):
+        self.owner = owner
+        self.repo = repo
+
+    def _request(self, href, params=None):
+        token = os.environ['TRAVIS_TOKEN']
+        headers = {
+            "Travis-API-Version": "3",
+            "Authorization": f"token {token}"
+        }
+
+        if params is None:
+            params = dict()
+        
+        resp = requests.get(self.url + href, headers=headers, params=params)
+        return json.loads(resp.text)
+
+    @max_items()
+    def builds(self, branch=None):
+        repo_slug = urllib.parse.quote_plus(f"{self.owner}/{self.repo}")
+        href = f"/repo/{repo_slug}/builds"
+
+        params = {}
+        if branch is not None:
+            params["branch.name"] = branch
+    
+        while href:
+            resp = self._request(href, params)
+    
+            for b in resp['builds']:
+                yield b
+    
+            try:
+                href = resp['@pagination']['next']['@href']
+            except KeyError:
+                raise StopIteration
+
+    def builds_for_branch(self, branch):
+        return self.builds(branch)
+
+    def builds_for_commit(self, branch, sha):
+        for b in self.builds(branch):
+            if b['commit']['sha'] == sha:
+                yield b
+
+
+class GitLab:
+    url = "https://allspark.dev.data.humancellatlas.org/api/v4"
+
+    def __init__(self, owner, repo):
+        self.owner = owner
+        self.repo = repo
+
+    def _request(self, href, params=None):
+        token = os.environ['GITLAB_TOKEN']
+        headers = {"PRIVATE-TOKEN": token}
+
+        if params is None:
+            params = dict()
+        
+        req_url = self.url + href
+        while req_url:
+            page = requests.get(req_url, headers=headers, params=params)
+            yield json.loads(page.text)
+            req_url = parse_paging_header(page.headers)
+
+    @max_items()
+    def builds(self, branch):
+        repo_slug = urllib.parse.quote_plus(f"{self.owner}/{self.repo}")
+        for page in self._request(f"/projects/{repo_slug}/jobs", {"ref_name": branch}):
+            for c in page:
+                yield c
+
+    @max_items()
+    def builds_for_commit(self, branch, sha):
+        for b in self.builds(branch):
+            if b['commit']['id'] == sha:
+                yield b
+
+    @max_items()
+    def commits(self, branch):
+        repo_slug = urllib.parse.quote_plus(f"{self.owner}/{self.repo}")
+        for page in self._request(f"/projects/{repo_slug}/repository/commits", {"ref_name": branch}):
+            for c in page:
+                yield c
+
+    def latest_commit(self, branch):
+        for c in self.commits(branch):
+            return c['id']
+
+
+def should_deploy_dev():
+    owner = "HumanCellAtlas"
+    repo = "data-store"
+    branch = "master"
+
+    GH = GitHub(owner, repo)
```

Comment:
Suggest: Not a constant.

Reply:
But there aren't any constants in Python.. changed